src/bosh_azure_cpi/lib/cloud/azure/vm_manager.rb

```diff
@@ -513,8 +549,17 @@ def default_update_domain_count
       @azure_config['environment'] == ENVIRONMENT_AZURESTACK ? 1 : 5
     end
 
-    def get_or_create_availability_set(resource_group_name, availability_set_params)
-      availability_set_name = availability_set_params[:name]
+    def get_or_create_availability_set(resource_group_name, availability_set_name, vm_properties, location)
+      return nil if availability_set_name.nil?
+
+      availability_set_params = {
+        name: availability_set_name,
+        location: location,
+        tags: AZURE_TAGS,
+        platform_update_domain_count: vm_properties['platform_update_domain_count'] || default_update_domain_count,
+        platform_fault_domain_count: vm_properties['platform_fault_domain_count'] || default_fault_domain_count,
+        managed: @use_managed_disks
+      }
```

Comment:
Originally, `availability_set_params` is outside of the function `get_or_create_availability_set`. The reason:
Only `platform_update_domain_count` and `platform_fault_domain_count` in `vm_properties` are required. So, there's no need to use `vm_properties` as the parameter of `get_or_create_availability_set`.

Reply:
Got your point, thanks. however, vm_properties is widely used in many places, I think it is ok. And putting the code here can make vm_manager.create simpler.

Reply:
please help review the PR i sent, the damn vm_properties is killed there.